xlators/features/shard/src/shard.c

```diff
@@ -5634,13 +5651,55 @@ shard_common_inode_write_post_lookup_shards_handler(call_frame_t *frame,
     }
 
     if (local->create_count) {
-        shard_common_resume_mknod(frame, this,
-                                  shard_common_inode_write_post_mknod_handler);
+        local->call_count = local->create_count;
+        shard_count = local->create_count;
+        num_shards_ex_base = local->num_blocks - 1;
+
+        if (local->create_count == num_shards_ex_base)
+            start_block = 0;
+        else
+            start_block = local->num_blocks - local->create_count + 1;
+
+        while (shard_count) {
+            if (shard_count < 5) {
```

Comment:
5 is a magic number?

Reply:
Apologies. I just kept it for testing. I will open up the patch for review once the complete implementation is done